### DISCUSSION TITLE
Unable to get AAD login to function, get two different errors

### DISCUSSION THREAD

**Discussion by mntsupport** [requarks/wiki#3470] [Help / Questions]
**Question**
I'm trying to set up AAD Auth and I'm hitting a snag. I followd the KB article when creating the Azure app registration and when I try to log in I get **AADSTS50011: The reply URL specified in the request does not match the reply URLs configured for the application:** This is using the https://YOUR-WIKI.DOMAIN.COM/login/azure/callback as the Redirect URI.

I noticed in the Admin > Auth> AAD settings in Wiki.js that there is a reference Redirect URI and when I put that one in Azure in place of the URL from the KB I get this error: 
**Oops, something went wrong...
Invalid email / username or password.**

How do I get this working?

**Host Info (please complete the following information):**
OS: Ubuntu 18.04, Docker
Wiki.js version: 2.5.159
Database engine: postgres 11.8

**Comment by yaegashi** [comment]
If you are using Chrome browser, it could be affected by [recent changes in SameSite cookies](https://blog.chromium.org/2020/02/samesite-cookie-changes-in-february.html).  You might want to test with other browsers like Firefox or Safari.

https://github.com/Requarks/wiki/pull/2567 is a workaround for this problem.  You need to run Wiki.js on an SSL host (or localhost for development) for it to work.

**Comment by mntsupport** [comment]
Got it, I'll be honest that implementing these changes seem a little over my skill level so I suppose I'll just wait until it hopefully makes it's way into the main build.

For visibility I also tried to use Firefox and I get the same error. It's like the callback URL is incorrect but I have no clue what it wants.

**Comment by colin-precisionos** [comment]
I seem to be getting the same thing as well.. AADSTS50011: The reply URL specified in the request does not match the reply URLs configured for the application.  I have spend hours going over the settings and everything looked correct for the redirect URL - assuming it's still the same format:
https://wiki.mywebsite.com/login/azure/callback

Fresh install in a linux: version 2.5.170, docker install.
I've also tried on Safari with the same result,  Is there anything else I can do to help the process?

**Comment by jeroenlandheer** [comment]
@colin-precisionos The documentation on this isn't really that clear, but if you look at the settings page where you configure the AAD logon, you'll see the correct redirect URI at the secion "Configuration Reference"

**Comment by jeroenlandheer** [comment]
@NGPixel Any idea when this PR will be implemented? (We're having issues with Chrome here as well.)

**Comment by colin-precisionos** [comment]
@jeroenlandheer  - that worked - thanks.  
Follow up
For those that get an **invalid user/password**, the system is getting confused between cookies for the Local and AAD authentication using the same email.  Clear your browser history and AAD will work

**Comment by avanguardigit** [comment]
> @jeroenlandheer - that worked - thanks.
> Follow up
> For those that get an **invalid user/password**, the system is getting confused between cookies for the Local and AAD authentication using the same email. Clear your browser history and AAD will work

I have the same problem.
I have try to clear the browser (crome and edge), same result
I have install firefox for the first time in this pc, same result
I try to use Iphone with safari, same result

what can I do?

**Comment by arnerb** [comment]
> Follow up
> For those that get an **invalid user/password**, the system is getting confused between cookies for the Local and AAD authentication using the same email. Clear your browser history and AAD will work

 # Quick message for your wiki users 
 _(until this is merged - no coding required - simply replace YOUR.WIKI.DOMAIN)_

#
Invalid username / password (Chrome)

- Fix is coming in Q2 2021 (already merged in dev, see this [post](https://github.com/Requarks/wiki/issues/2612#issuecomment-715670324)). Workaround till then: 

  - Delete **connect.sid** cookie found here:

    chrome://settings/cookies/detail?site=YOUR.WIKI.DOMAIN
    (**copy paste above into chrome**)

    Note: For easy access, you can also create a bookmark with Ctrl+D

  -   these also would work
      Clear browsing data > cookies > since: last time you logged in successfully
      Use incognito mode (manual login)
      Use another browser (firefox, safari,..)

**Comment by devksingh4** [comment]
I am not having luck using incognito browser either... anyone have any tips?

**Comment by iamgrimm** [comment]
I am also having this issue, has anyone gotten to the bottom of this?

**Comment by jeroenlandheer** [comment]
@iamgrimm Have you tried this? https://github.com/Requarks/wiki/issues/2612#issuecomment-755194820

**Comment by captainhook** [comment]
I have just encountered this issue on a fresh setup using Firefox. Clearing my wikijs domain's cookies from the browser fixed it, and I added the 'Callback URL / Redirect URI' listed in Wiki.js AAD authentication page to the Redirect URIs in AAD App.

**Comment by jomaroc** [comment]
On Wiki.js documentation show this step:

`3. Enter a Name (e.g. Wiki.js) and enter the Redirect URI in the following format: https://YOUR-WIKI.DOMAIN.COM/login/azure/callback
`

But the value to configure is the **Callback URL / Redirect URI** defined on the **Configuration Reference** section on **your Azure Active Directory Strategy**

something like this:
`https://YOURDOMAINNAME.COM/login/4XXXX9c1-a9XX-XXX3-8XX4-XX5baXXXXdf7/callback`
